utils/src/main/java/software/amazon/awssdk/utils/async/DelegatingBufferingSubscriber.java

```diff
@@ -43,17 +44,16 @@ public class DelegatingBufferingSubscriber extends AbstractFlatteningSubscriber<
      */
     private final StoringSubscriber<ByteBuffer> storage = new StoringSubscriber<>(Integer.MAX_VALUE);
 
-    public DelegatingBufferingSubscriber(long maximumBuffer, Subscriber<? super ByteBuffer> subscriber) {
-        super(subscriber);
-        this.maximumBuffer = maximumBuffer;
+    protected DelegatingBufferingSubscriber(Long maximumBufferInBytes, Subscriber<? super ByteBuffer> delegate) {
+        super(Validate.notNull(delegate, "delegate must not be null"));
+        this.maximumBufferInBytes = Validate.notNull(maximumBufferInBytes, "maximumBufferInBytes msut not be null");
     }
 
     @Override
     public void onNext(ByteBuffer item) {
-        if (currentlyBuffered.get() + item.remaining() > maximumBuffer) {
+        if (currentlyBuffered.get() > 0) {
```

Comment:
Don't we need to remove `flushStorageToDelegate` from here as well?

Should we make onNext final in `BaseSubscriberAdapter` since handleStateUpdate is supposed to be handled in the base class?

Reply:
still needed to overwrite onNext to fix input stream transformer